src/decorator/CircuitBreakerDecorator.ts

```diff
@@ -0,0 +1,39 @@
+import { CircuitBreaker } from '../CircuitBreaker';
+import { CircuitBreakerState } from '../CircuitBreakerState';
+
+type CBConfig = {
+    failureThreshold: number;
+    recoveryTimeout: number;
+    observers?: ((previousState: CircuitBreakerState, currentState: CircuitBreakerState) => void)[];
+};
+
+/**
+ * CircuitBreakerDecorator is a decorator that wraps all the methods of a class with the wrapFunction from CircuitBreaker instance.
+ * @param {CBConfig} config
+ * @return {ClassDecorator}
+ * @example
+ *
+ * @CircuitBreakerDecorator({
+ *    failureThreshold: 5,
+ *    recoveryTimeout: 3000,
+ *  })
+ *  class MyClass {...}
+ *
+ */
+export function CircuitBreakerDecorator(config: CBConfig): ClassDecorator {
+    const { failureThreshold, recoveryTimeout } = config;
```

Comment:
Why not extracting `oberservers` here so that we read config object in only one place?